porcelain/miner_test.go

```diff
@@ -22,6 +23,103 @@ import (
 	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
 )
 
+type minerCreate struct {
+	config  *cfg.Config
+	require *require.Assertions
+	wallet  *wallet.Wallet
+}
+
+func newMinerCreate(require *require.Assertions) *minerCreate {
+	repo := repo.NewInMemoryRepo()
+	backend, err := wallet.NewDSBackend(repo.WalletDatastore())
+	require.NoError(err)
+	return &minerCreate{
+		config:  cfg.NewConfig(repo),
+		wallet:  wallet.New(backend),
+		require: require,
+	}
+}
+
+func (mpc *minerCreate) ConfigGet(dottedPath string) (interface{}, error) {
+	return mpc.config.Get(dottedPath)
+}
+
+func (mpc *minerCreate) ConfigSet(dottedPath string, paramJSON string) error {
+	return mpc.config.Set(dottedPath, paramJSON)
+}
+
+func (mpc *minerCreate) GetAndMaybeSetDefaultSenderAddress() (address.Address, error) {
+	return wallet.NewAddress(mpc.wallet)
+}
+
+func (mpc *minerCreate) MessageSend(ctx context.Context, from, to address.Address, value *types.AttoFIL, gasPrice types.AttoFIL, gasLimit types.GasUnits, method string, params ...interface{}) (cid.Cid, error) {
+	return cid.Cid{}, nil
+}
+
+func (mpc *minerCreate) MessageQuery(
+	ctx context.Context,
+	optFrom,
+	to address.Address,
+	method string,
+	params ...interface{},
+) ([][]byte, *exec.FunctionSignature, error) {
+	signature := &exec.FunctionSignature{
+		Params: nil,
+		Return: []abi.Type{abi.Address},
+	}
+	val := abi.Value{
+		Type: abi.Address,
+		Val:  address.Address{},
+	}
+	ret, err := val.Serialize()
+	mpc.require.NoError(err)
+	return [][]byte{ret}, signature, nil
+}
+
+func (mpc *minerCreate) MessageWait(ctx context.Context, msgCid cid.Cid, cb func(*types.Block, *types.SignedMessage, *types.MessageReceipt) error) error {
+	return nil
+}
+
+func (mpc *minerCreate) NetworkGetPeerID() peer.ID {
+	return peer.ID("")
+}
+
+func (mpc *minerCreate) WalletFind(addr address.Address) (wallet.Backend, error) {
+	return mpc.wallet.Find(addr)
+}
+
+func (mpc *minerCreate) WalletGetPubKeyForAddress(addr address.Address) ([]byte, error) {
+	return mpc.wallet.GetPubKeyForAddress(addr)
+}
+
+func (mpc *minerCreate) WalletHasAddress(addr address.Address) bool {
+	return mpc.wallet.HasAddress(addr)
+}
+
+func TestMinerCreate(t *testing.T) {
```

Comment:
blocking: this test doesn't give us much assurance that the call is working. at the very least you could use the fake objects above to assert that things are hooked up properly eg wait receives the cid returned by send and that invoking the wait callback with no a zero exit code results in no error, as well as a test that zero exit code in the receipt results in an error. you could also check the side effects. i think we need to improve the success case and have at least one error.